lib/artist.rb

```diff
@@ -0,0 +1,46 @@
+
+
+class Artist
+  extend Concerns::Findable
+
+  attr_accessor :name
+  attr_reader :songs
+
+  @@all = []
+
+  def initialize(name)
+    @name = name
+    @songs = []
+  end
+
+  def self.all
+    @@all
+  end
+
+  def self.destroy_all
+    all.clear
+  end
+
+  def save
+    @@all << self
+  end
+
+  def self.create(name)
+    artist = new(name)
+    artist.save
+    artist
```

Comment:
If you have your `save` method return the instance it is saving vs. the `@@all` array, you could refactor this down to one line `new(name).save1